src/Container/Container.php

```diff
@@ -0,0 +1,111 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Libs\Container;
+
+use DI\FactoryInterface;
+use Invoker\InvokerInterface;
+use Psr\Container\ContainerInterface;
+use ReflectionObject;
+
+/**
+ * A mutable container wrapper.
+ */
+class Container extends \DI\Container implements MutableContainer, ContainerInterface, FactoryInterface, InvokerInterface {
+
+	/**
+	 * A reflection of the wrapped container.
+	 *
+	 * @var ReflectionObject|null
+	 */
+	protected $reflectionContainer = null;
+
+	/**
+	 * Wrap an existing container and create a new instance of this container.
+	 *
+	 * Items in the wrapped container are still resolved in this container.
+	 *
+	 * @param  \Psr\Container\ContainerInterface|null  $container
+	 *
+	 * @return \Tribe\Libs\Container\MutableContainer|\DI\Container|ContainerInterface|\DI\FactoryInterface|\Invoker\InvokerInterface
+	 */
+	public function wrap( ?ContainerInterface $container = null ) {
+		if ( $container ) {
+			$container = clone $container;
```

Comment:
Hrmm very good question. We would never get an incomplete container clone, but it's possible the properties that reference external objects still point to those, which is exactly what I want to avoid. The main reason I'm cloning here is so that I can make internal changes to the object without affecting the original container in any way. I don't want the mutations to ever affect the main container.

Reply:
Alright, I just tested this: Because we clone `$container` and later assign to: `$this->delegateContainer` in our mutable container, we are able to mutate the delegate, without affecting the original container's instance.

Now when I force clear the `$resolvedEntries` property in our mutable container or any other modifications, I'm guaranteeing I don't mess around with the original container instance at all, isolating any mutations to our container/the delegate container only.

The PHP-DI container builder basically allows wrapping a container to integrate with other containers, but because our container is already built, I just mimic the logic that's happening when the builder is creating the container as outlined below:

References:
1. https://php-di.org/doc/container-configuration.html#using-php-di-with-other-containers
2. [Wrap container with another in the builder](https://github.com/PHP-DI/PHP-DI/blob/master/src/ContainerBuilder.php#L305)
3. [Builder creates a new container with the wrapped container](https://github.com/PHP-DI/PHP-DI/blob/master/src/ContainerBuilder.php#L191)
4. [Comes in the Container constructor](https://github.com/PHP-DI/PHP-DI/blob/master/src/Container.php#L102)
5. [I just mimic that behavior, after cloning the container so we don't mutate that one at all](https://github.com/moderntribe/tribe-libs/pull/102/files#diff-b3061446b6923316c68d46e450315e4a900b1fd223576d2bafa009bbd417f97eR36)